### PR TITLE
feat: Avatar Component

### PR DESCRIPTION
**Pull request by ozairkhan2810**
Please review this PR.

Here is link for website
https://objective-pasteur-71db3d.netlify.app/index.html

### CODE REVIEW

File: css/style.css [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/* navbar styling for landing page and documentation page */
+.navbar {
+  z-index: 10;
+  background: var(--bg);
+  /* margin-top: 2rem; */
```
Comment (harshitadaswani) [inline css/style.css]: remove commented code if not in use

File: css/style.css [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+/* sidebar styling for documentation page */
+.sidebar {
+  width: 12.5rem;
+  padding: 0 1rem;
+  font-size: 1.3rem;
+  position: fixed;
+  /* top: 0; */
+  height: 100vh;
+  /* border-right: 2px solid rgb(218, 202, 202); */
+  /* margin-top: 2rem; */
```
Comment (harshitadaswani) [inline css/style.css]: remove commented code

File: lib/base/reset.css [hunk 3]
```diff
@@ -9,6 +9,8 @@ html, body {
     padding: 0;
     font-size: 100%;
     background: var(--bg);
+    /* border: 5px solid var(--fg); */
+    /* height: 100vh; */
```
Comment (harshitadaswani) [inline lib/base/reset.css]: remove commented code

File: lib/base/reset.css [hunk 4]
```diff
@@ -9,6 +9,8 @@ html, body {
     padding: 0;
     font-size: 100%;
```
Comment (harshitadaswani) [inline lib/base/reset.css]: use `font-size: inherit` instead

File: lib/base/utilities.css [hunk 5]
```diff
@@ -0,0 +1,67 @@
+.container {
+    max-width: 960px;
```
Comment (harshitadaswani) [inline lib/base/utilities.css]: use rem or % units

File: lib/base/variables.css [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  --bg1: #3c3836;
+  --bg2: #504945;
+  --bg3: #665c54;
+  --bg4: #7c6f64;
+
+  --fg: #fbf1c7;
+  --fg1: #ebdbb2;
+  --fg2: #d5c4a1;
+  --fg3: #bdae93;
+  --fg4: #a89984;
```
Comment (harshitadaswani) [inline lib/base/variables.css:14]: use descriptive color names for reviewer to understand

File: lib/base/variables.css [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  --bg1: #ebdbb2;
+  --bg2: #d5c4a1;
+  --bg3: #bdae93;
+  --bg4: #a89984;
+
+  --fg: #282828;
+  --fg1: #3c3836;
+  --fg2: #504945;
+  --fg3: #665c54;
+  --fg4: #7c6f64;
```
Comment (harshitadaswani) [inline lib/base/variables.css:49]: same as above

File: lib/base/variables.css [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  --gray: #928374;
+
+  --red-dim: #cc2412;
+  --green-dim: #98971a;
+  --yellow-dim: #d79921;
+  --blue-dim: #458598;
+  --purple-dim: #b16286;
+  --aqua-dim: #689d6a;
+  --orange-dim: #d65d0e;
+  --gray-dim: #7c6f64;
```
Comment (harshitadaswani) [inline lib/base/variables.css:67]: why you have used same color codes again?

File: lib/components/avatar.css [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    height: 4.5rem;
+    width: 4.5rem;
+}
+
+.av-sm,
+.av-md, 
+.av-lg, 
+.av-xl {
+    border-radius: 100%;
+}
```
Comment (harshitadaswani) [inline lib/components/avatar.css]: you can add .av in the common class code and remove the `border-radius` from `.av class` so the code is not redundant

File: lib/components/avatar.css [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+.av-md {
+    height: 4rem;
+    width: 4rem;
+}
+
+.av-lg {
+    height: 4.5rem;
+    width: 4.5rem;
+}
```
Comment (harshitadaswani) [inline lib/components/avatar.css:27]: use comma selector as the properties are common with `.av` class

File: lib/components/avatar.css [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    background: lightgray;
+}
+
+.rel {
+    position: relative;
+}
+
+/* .icon {
+    font-size:2rem;
+} */
```
Comment (harshitadaswani) [inline lib/components/avatar.css]: remove commented code

**Review by harshitadaswani — CHANGES_REQUESTED** [review]
Refactor the code according to the suggestion given otherwise, the code looks good to merge.

**Comment by rushikesh3299** [comment]
Please follow suggestions given by harshita & if you want to make documentation site responsive you can try that from start otherwise after code has grown it will become difficult for you.

File: css/style.css (hunk 1 above)
Comment (ozairkhan2810) [reply css/style.css]: Done

File: css/style.css (hunk 2 above)
Comment (ozairkhan2810) [reply css/style.css]: Done

File: lib/base/reset.css (hunk 3 above)
Comment (ozairkhan2810) [reply lib/base/reset.css]: Done

File: lib/base/reset.css (hunk 4 above)
Comment (ozairkhan2810) [reply lib/base/reset.css]: Done.
Thanks for this suggestion.

File: lib/base/utilities.css (hunk 5 above)
Comment (ozairkhan2810) [reply lib/base/utilities.css]: Done

File: lib/base/variables.css (hunk 6 above)
Comment (ozairkhan2810) [reply lib/base/variables.css:14]: Those colors are for background(bg) and foreground(fg) with shades in numbers.
Hope you got it.
Thanks.

File: lib/base/variables.css (hunk 7 above)
Comment (ozairkhan2810) [reply lib/base/variables.css:49]: Those colors are for background(bg) and foreground(fg) with shades in numbers.
Hope you got it.
Thanks.

File: lib/base/variables.css (hunk 8 above)
Comment (ozairkhan2810) [reply lib/base/variables.css:67]: One is for light mode and another for dark mode.

File: lib/components/avatar.css (hunk 9 above)
Comment (ozairkhan2810) [reply lib/components/avatar.css]: Done.
Thanks for pointing this.

File: lib/components/avatar.css (hunk 10 above)
Comment (ozairkhan2810) [reply lib/components/avatar.css:27]: Done.
Changed properties of .av class to other value.
Thanks

File: lib/components/avatar.css (hunk 11 above)
Comment (ozairkhan2810) [reply lib/components/avatar.css]: Removed.

**Review by Raghav888 — APPROVED** [review]
Looks good.

**Review by harshitadaswani — APPROVED** [review]
yeah you can go ahead

**Pull request merged**